sunpy/map/maputils.py

```diff
@@ -153,21 +175,59 @@ def contains_full_disk(smap):
     within the field of the view of the instrument (although no emission
     from the disk itself is present in the data.)
     """
-    # Calculate all the edge pixels
-    edges = map_edges(smap)
-    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]]))
-    x = [p[0] for p in edge_pixels] * u.pix
-    y = [p[1] for p in edge_pixels] * u.pix
-
-    # Calculate the edge of the world
-    edge_of_world = smap.pixel_to_world(x, y)
-
-    # Calculate the distance of the edge of the world in solar radii
-    coordinate_angles = np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2)
-
-    # Test if all the edge pixels are more than one solar radius distant
-    # and that the whole map is not all off disk.
-    return np.all(coordinate_angles > solar_angular_radius(edge_of_world)) and ~is_all_off_disk(smap)
+    # Get the edge pixels
+    top_, bottom, left_hand_side, right_hand_side = map_edges(smap)
+
+    # Coordinates of the edges
+    horizontal1 = smap.pixel_to_world(*_convert_pixels(top_))
+    horizontal2 = smap.pixel_to_world(*_convert_pixels(bottom))
+    vertical1 = smap.pixel_to_world(*_convert_pixels(left_hand_side))
+    vertical2 = smap.pixel_to_world(*_convert_pixels(right_hand_side))
+
+    # The radius of the Sun
+    radius = smap.rsun_obs
+
+    # Determine which of the inputs are top and bottom edges of the map
+    top = None
+    bot = None
+    h1 = horizontal1.Ty
+    if np.all(h1 > radius):
+        top = h1
+    elif np.all(h1 < -radius):
+        bot = h1
+
+    h2 = horizontal2.Ty
+    if np.all(h2 > radius):
+        top = h2
+    elif np.all(h2 < -radius):
+        bot = h2
+
+    # If neither edge lies further away than a radius then the map
+    # cannot contain the full disk.
+    if top is None or bot is None:
+        return False
+
+    # Determine which of the inputs are the left and right hand edges of the map
+    lhs = None
+    rhs = None
+    v1 = vertical1.Tx
+    if np.all(v1 > radius):
+        rhs = v1
+    elif np.all(v1 < -radius):
+        lhs = v1
+
+    v2 = vertical2.Tx
+    if np.all(v2 > radius):
+        rhs = v2
+    elif np.all(v2 < -radius):
+        lhs = v2
+
```

Comment:
These basically repeat but changed out for a different variable. There should be a neater way to write this but I don't have an answer at the moment.

Reply:
Yeah, I agree.  I wrote the function quickly and in a long-winded way to support the hackweek.